mat64/io_test.go

```diff
@@ -39,3 +49,36 @@ func TestDenseRW(t *testing.T) {
 		}
 	}
 }
+
+func TestVectorRW(t *testing.T) {
+	for i, test := range []*Vector{
+		NewVector(0, []float64{}),
```

Comment:
is it really something that can happen?
https://github.com/gonum/matrix/blob/master/mat64/vector.go#L35 should insure that `n == len(data)`, always.

or are you wondering whether we should make sure the panic, when unmarshalling a corrupted binary file, is crystal clear?